src/viper/trans.ml

```diff
@@ -272,6 +283,8 @@ and unit' (u : M.comp_unit) : prog =
     ) is''' in
     let perm_def = !!! (body.at) (InvariantI("$Perm", perm body.at)) in
     let inv_def = !!! (body.at) (InvariantI("$Inv", adjoin ctxt'' (conjoin invs body.at) !(ctxt.ghost_conc))) in
+    (* given is', compute ghost_is *)
+    let ghost_is = List.map (fun mk_i -> mk_i ctxt') !(ctxt.ghost_items) in
```

Comment:
Okay, just if it's not need anymore move it back.
Otherwise it looks a bit weird here, because it uses `ctxt'` and `ctxt` where even `ctxt''` is available.

Reply:
Moved it back